### PR TITLE
speech: add livecaption

### PR DESCRIPTION
**Pull request by rakyll**

### CODE REVIEW

File: speech/livecaption/livecaption.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		option.WithEndpoint("speech.googleapis.com:443"),
+		option.WithScopes("https://www.googleapis.com/auth/cloud-platform"),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer conn.Close()
+
+	// [START speech_streaming_mic_recognize]
+	stream, err := speech.NewSpeechClient(conn).StreamingRecognize(ctx)
```
Comment (broady) [inline speech/livecaption/livecaption.go]: https://godoc.org/cloud.google.com/go/speech/apiv1beta1#NewClient

File: speech/livecaption/livecaption.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	// send the initial configuration message.
```
Comment (broady) [inline speech/livecaption/livecaption.go]: sentence casing, please (and in the two comments below)

`// Send the initial configuration message.`

File: speech/livecaption/livecaption.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	go func() {
+		// pipe stdin to the API
+		buf := make([]byte, 1024)
+		for {
+			n, err := os.Stdin.Read(buf)
+			if err == io.EOF {
+				return // nothing else to pipe, kill this goroutine
+			}
+			if err != nil {
+				log.Printf("Reading stdin error: %v", err)
```
Comment (broady) [inline speech/livecaption/livecaption.go]: "Could not read from stdin"

File: speech/livecaption/livecaption.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+				continue
+			}
+			if err = stream.Send(&speech.StreamingRecognizeRequest{
+				StreamingRequest: &speech.StreamingRecognizeRequest_AudioContent{
+					AudioContent: buf[:n],
+				},
+			}); err != nil {
+				log.Printf("Sending audio error: %v", err)
```
Comment (broady) [inline speech/livecaption/livecaption.go]: "Could not send audio:"

File: speech/livecaption/livecaption.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+		}
+	}()
+
+	for {
+		resp, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatalf("stream.Recv error: %v", err)
```
Comment (broady) [inline speech/livecaption/livecaption.go]: `"stream.Recv: %v"`

unless there's a better way to describe what might have happened

File: speech/livecaption/livecaption.go [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+		}
+		if err := resp.Error; err != nil {
+			log.Fatalf("Recieved error resp: %v", err)
```
Comment (broady) [inline speech/livecaption/livecaption.go]: "Could not recognize"?

File: speech/livecaption/livecaption.go [hunk 7]
```diff
@@ -53,23 +58,23 @@ func main() {
...
 
 	go func() {
-		// pipe stdin to the API
+		// Pipe stdin to the API.
 		buf := make([]byte, 1024)
 		for {
 			n, err := os.Stdin.Read(buf)
 			if err == io.EOF {
-				return // nothing else to pipe, kill this goroutine
+				return // Nothing else to pipe, returns from this goroutine.
```
Comment (broady) [inline speech/livecaption/livecaption.go]: just `return`, not `returns`

**Comment by broady** [comment]
LGTM, one nit

File: speech/livecaption/livecaption.go (hunk 7 above)
Comment (rakyll) [inline speech/livecaption/livecaption.go]: Done

**Comment by broady** [comment]
👍

**Pull request merged**